openedx/frontend-app-admin-portal
# feat: add assigned table to LCM activity tab

# Description

Tickets: 
* [ENT-7611](https://2u-internal.atlassian.net/browse/ENT-7611)
* [ENT-7610](https://2u-internal.atlassian.net/browse/ENT-7610) 

Contents:
* Adds "Assigned" table UI under the Learner Credit Management "Activity" page route with API integration.
* Implements API integration with assignments REST API to populate the table data.
* Implements API integration with subsidy access policy REST API to retrieve the metadata of an individual budget, which returns an assignment configuration UUID to be called by the assignments table.
* Implements a `useSubsidyAccessPolicy` hook that utilizes `useQuery` to act as a state manager for the subsidy access policy from the detail API. This is possible by expanding the `staleTime` on queries. This allows us to expose commonly used metadata across subcomponents to avoid prop-drilling and context (e.g., making the `subsidyAccessPolicy` metadata available, including its `catalog_uuid` in the "Catalog" tab).
* Ensures "Catalog" tab and "Assigned" table are only exposed in the UI if the viewed budget is assignable, based on its budget type (i.e., `AssignedLearnerCreditAccessPolicy`).

### With feature flag ENABLED

![image](https://github.com/openedx/frontend-app-admin-portal/assets/2828721/eb5d00b6-80dc-4c87-b66e-f6c2a8c1a17b)

![image](https://github.com/openedx/frontend-app-admin-portal/assets/2828721/25c838ec-63d4-4907-855d-c16e5e0080cb)

### With feature flag DISABLED

![image](https://github.com/openedx/frontend-app-admin-portal/assets/2828721/f13fca49-3655-45fe-82d1-4670605749b1)

## Testing instructions

1. Ensure `enterprise.top_down_assignment_real_time_lcm` Waffle-flag is enabled within the LMS for your user.
2. Find a subsidy access policy UUID of an assignable budget.
3. Visit `localhost:1991/:enterpriseSlug/admin/learner-credit/:subsidyAccessPolicyUUID` to load the budget detail page.
4. You should see:
  - Subsidy access policy's display name, if any.
  - Initial "Assigned" table with any allocated/errored assigments.
  - "Spent" table (may be empty if you don't have analytics API data).

# For all changes


# Only if submitting a visual change
